gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -410,6 +440,66 @@ ChannelCredentials createMtlsChannelCredentials() throws IOException, GeneralSec
     return null;
   }
 
+  @VisibleForTesting
+  ChannelCredentials createMtlsToS2AChannelCredentials(
+      InputStream trustBundle, InputStream privateKey, InputStream certChain) throws IOException {
+    if (trustBundle == null || privateKey == null || certChain == null) {
+      return null;
+    }
+    return TlsChannelCredentials.newBuilder()
+        .keyManager(privateKey, certChain)
+        .trustManager(trustBundle)
+        .build();
+  }
+
+  ChannelCredentials createS2ASecuredChannelCredentials() {
```

Comment:
Also, it doesn't have to be as javadocs, but could you add some small comments to the top that describes the logic in the method?

Reply:
Done in https://github.com/googleapis/sdk-platform-java/pull/3326/commits/e4565f49eefada0d4a760098c2b4cba1e2cefee9